nixos/modules/misc/version.nix

```
@@ -41,7 +41,7 @@ let
       IMAGE_VERSION = optionalString (config.system.image.version != null) config.system.image.version;
       VARIANT = optionalString (cfg.variantName != null) cfg.variantName;
       VARIANT_ID = optionalString (cfg.variant_id != null) cfg.variant_id;
-      DEFAULT_HOSTNAME = config.networking.fqdnOrHostName;
+      DEFAULT_HOSTNAME = optionalString (config.networking.hostName != "") config.networking.fqdnOrHostName;
```

I may be misunderstanding the manual page, but I think both implementations are incorrect. This setting should not deviate from the distribution default, which is `config.system.nixos.distroId`. An user setting a different hostname should not alter the distribution default.
```suggestion
      DEFAULT_HOSTNAME = config.system.nixos.distroId;
```